incidents/publish.py

```diff
@@ -0,0 +1,145 @@
+
+import json
+import logging
+
+from botocore.client import BaseClient, ClientError
+
+
+OPERATORS = {'VT': 'Avanti West Coast',
+             'CC': 'c2c',
+             'CS': 'Caledonian Sleeper',
+             'CH': 'Chiltern Railways',
+             'XC': 'CrossCountry',
+             'EM': 'East Midlands Railway',
+             'XR': 'Elizabeth Line',
+             'ES': 'Eurostar',
+             'GC': 'Grand Central',
+             'LF': 'Grand Union Trains',
+             'GN': 'Great Nothern',
+             'GW': 'Great Western Railway',
+             'LE': 'Greater Anglia',
+             'HX': 'Heathrow Express',
+             'HT': 'Hull Trains',
+             'IL': 'Island Lines',
+             'LO': 'London Overground',
+             'LT': 'London Underground',
+             'LN': 'London Northwestern Railway',
+             'GR': 'LNER',
+             'LD': 'Lumo',
+             'ME': 'Merseyrail',
+             'NT': 'Northern Trains',
+             'SR': 'ScotRail',
+             'SE': 'Southeastern',
+             'SN': 'Southern',
+             'SW': 'South Western Railway',
+             'SP': 'Swanage Railway',
+             'SX': 'Stansted Express',
+             'TP': 'TransPennine Express',
+             'AW': 'Transport for Wales',
+             'TL': 'Thameslink',
+             'WR': 'West Coast Railways',
+             'LM': 'West Midlands Trains',
+             'GX': 'Gatwick Express',
+             'WM': 'West Midlands Railway'}
+
+
+def convert_toc_code_to_name(toc_code: list[str]) -> list[str]:
+    """Takes in a list of operator codes
+    and returns list of operator names."""
+
+    return [OPERATORS.get(code, " ") for code in toc_code]
+
+
+def get_message_attributes(operators: list) -> dict:
+    """Takes in a list of operator names and
+      appends to the message attributes dictionary."""
+
+    message_attributes = {}
+
+    message_attributes['operators'] = {
+        "DataType": "String.Array",
+        "StringValue": json.dumps(operators)
+    }
+    return message_attributes
+
+
+def publish_multi_message_with_attributes(sns_client: BaseClient, topic_arn: str,
+                                          message_subject: str, sms_message: str,
+                                          email_message: str, message_attributes: dict):
+    """Publishes a multi-format message with attributes to a topic."""
+
+    try:
+        message = {
+            "default": "default message",
+            "sms": sms_message,
+            "email": email_message,
+        }
+
+        response = sns_client.publish(
+            TopicArn=topic_arn,
+            Message=json.dumps(message),
+            MessageAttributes=message_attributes,
+            Subject=message_subject,
+            MessageStructure="json"
+        )
+        message_id = response["MessageId"]
+
+        logging.info(
+            "Published multi-format message to topic %s.", topic_arn)
+
+    except ClientError:
```

Comment:
Good use of the try except clause - Using a specific error to help with debugging. Well done!